platform/darwin/src/MGLStyleAttributeValue.h

```diff
@@ -0,0 +1,31 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLTypes.h"
+
+typedef struct {
+    float red;
+    float green;
+    float blue;
+    float alpha;
+} MGLAttributeColor;
```

Comment:
Ah, I see that you’re already extending NSColor and UIColor that way. So `-[MGLStyleAttributeValue colorValue]` should return an NSColor or UIColor, and whatever calls `-colorValue` would use `-[NSColor(MGLAdditions) mbgl_color]` to convert to an `mbgl::Color` for use with core code.